frontend-react/docs/proposals/0003-uswds-react-components.md

```diff
@@ -0,0 +1,251 @@
+- On top of the base USWDS library, we also use [Trussworks React component library](https://github.com/trussworks/react-uswds). This adds another level of rigidity to the stack since we aren't interacting with the base USWDS library, we're only exposed to whatever `props` trussworks exposes for us.
+- From-the-ground-up customized components while offering complete control, could move us too far off the standardized, accessible components offered by the USWDS + would be quite a lot of work to create and maintain.
+
+## Decision Drivers
+
+- Recent experience adding a simple addition to USWDS showed that we can't rely on them to update their library according to our needs.
+- Our `<Table />` component is becoming increasing complex and to be realized fully, requires us to deviate significantly from the USWDS base `<Table />` component.
+
+## Considered Options
+
+In a very similar vein to [Nava's previous ADR](https://github.com/navapbc/template-application-nextjs/blob/0f58517add316235e7c187bd00a820e91d4221c9/docs/decisions/app/0004-uswds-in-react.md) on the subject, there are 3 options available to us.
+
+For each of the following options, I will build a simple feature on top of the concept of the USWDS `<Table />` component: Adding a sticky header.
+
+---
+
+<a name="option-one"></a>
+### Option 1: Wrap Trussworks USWDS React Components
+
+#### Description
+
+The Trussworks React component library is, in an ideal world, all we would need. As taken from the [Nava ADR](https://github.com/navapbc/template-application-nextjs/blob/0f58517add316235e7c187bd00a820e91d4221c9/docs/decisions/app/0004-uswds-in-react.md#use-the-existing-open-source-react-uswds-library), Trussworks maintains this repo, is a trusted vendor in the space, and they keep their library very up-to-date.
+
+Wrapping their components provides all of their underlying work (latest USWDS version, adhere strictly to USWDS code design, etc) for free.
+
+#### Pros
+
+- Trussworks handles turning the USWDS base library into exposed React components.
+- Extremely actively maintained with an associated [Trussworks Storybook](https://trussworks.github.io/react-uswds/) which would make diffing our extended components very easy.
+- Follows our currently implemented solutions of extending components, so least overhead in terms of code changes.
+- Ideal for stylistic changes to Trussworks React components. Can simply use the scoped `.module` syntax to create new dynamic styles.
+- No need to add styles to our convoluted overrides file `_uswds_overrides.scss` anymore.
+
+#### Cons
+- Not ideal for complex functional extensions of React components.
+- Can get convoluted quickly since we're essentially wrapping a wrapper.
+- Potentially easy breaking changes as we'd be two levels from USWDS.
+- Still need to manually override USWDS styles on occasion using `!important`.
```

Comment:
Related but not dependent on our decision in this ADR: I think we should avoid `!important`s as much as possible and instead try to find a scalable pattern that'll allow us to override USWDS defaults through specificity rather than the `!important` steamroller.  Basically, my feeling is that we should be keeping specificity as low as possible so styles can be more easily overridden downstream.

Reply:
I totally agree. I would _much_ prefer if we could just use CSS specificity to overwrite any required CSS changes. I guess I was just going showcasing the worst case scenario (and should've been more explicit about that) if we had a direct collision and needed to use `!important`. Can add a summary of this conversation to the ADR.

Reply:
Makes sense to me!